src/TLuaInterpreter.cpp

```diff
@@ -4210,7 +4210,15 @@ int TLuaInterpreter::setRoomIDbyHash(lua_State* L)
         hash = lua_tostring(L, 2);
     }
     Host& host = getHostFromLua(L);
-    host.mpMap->mpRoomDB->hashTable[QString(hash.c_str())] = id;
+    QString hashStr = QString(hash.c_str());
+    if ( host.mpMap->mpRoomDB->roomIDToHash.contains(id) ){
```

Comment:
:thinking: @vadi2 should we consider whether we should have a separate function to *clear* a roomId \<==> hash linkage and then reject the creation in this function of such an association if ***either*** the roomId ***or*** the hash are already present in the pair of `QMap`s (or `QHash`es if another of my suggestions are acted on)?

Reply:
I'm not sure, I don't use room hashes. Let's wait for a use-case?

Reply:
As a related matter - how do you *remove* a hash/room ID if setting an empty hash or non-existent room id does not work to clear the corresponding entry? :worried:

Reply:
The user doesn't. Not directly. A user can delete the room with the corresponding ID (or create it and then delete it if it doesn't exist).
Otherwise, we now correctly remove the old hash if a new one is set.
Before, it just leaked.